torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -6898,7 +6947,20 @@ def test_ddp_profiling_torch_profiler(self):
             cpu_act = torch.profiler.ProfilerActivity.CPU
             cuda_act = torch.profiler.ProfilerActivity.CUDA
             torch_profiler_ctx = torch.profiler.profile(activities=[cpu_act, cuda_act])
-            self._test_ddp_profiling(profiler_ctx=torch_profiler_ctx)
+            prof = self._test_ddp_profiling(profiler_ctx=torch_profiler_ctx)
+
+            if dist.get_backend() == "nccl":
+                nccl_meta = self._validate_profiler_nccl_meta(prof)
+                self.assertEqual(len(nccl_meta["allreduce"]), 2)
+                self.assertEqual(len(nccl_meta["wait"]), 1)
+
+                # check allreduce message sizes
+                a0 = nccl_meta["allreduce"][0]
+                self.assertEqual(a0["Out msg nelems"], 100, msg=f"{a0}")
+                self.assertEqual(a0["dtype"], "Float", msg=f"{a0}")
+                a1 = nccl_meta["allreduce"][1]
+                self.assertEqual(a1["Out msg nelems"], 1, msg=f"{a1}")
+                self.assertEqual(a1["dtype"], "Int", msg=f"{a1}")
```

Comment:
nit: is there a reason for `_validate_profiler_nccl_meta` to happen at a different place than the rest below it?

nit: is there a way not to hard code those expected values? Can they be parametrized with _test_ddp_profiling?

Reply:
_validate_profiler_nccl_meta() is mainly doing some sanity checking but the actual expected values are test specific. (TBH this is the only test that checks these statistics though) I can rename to _sanity_check_check_profiler_nccl_meta() ?

> nit: is there a way not to hard code those expected values? Can they be parametrized with _test_ddp_profiling?

not sure what you mean by parametrize in this case?  Since we have an expected set of collectives in test_dpp_profiling() we know the values to check against.